src/ui/SimpleFilter/SimpleFilter.ts

```diff
@@ -0,0 +1,188 @@
+            return null;
+        }),
+    };
+
+    public checkboxes: {checkbox: Checkbox, value: string} [] =  [];
+    public checkboxContainer: Dom;
+    private expanded = false;
+
+    constructor(public element: HTMLElement, public options: ISimpleFilterOptions, public bindings?: IComponentBindings){
+        super(element, SimpleFilter.ID, bindings);
+        this.type = 'SimpleFilter';
+        this.options = ComponentOptions.initComponentOptions(element, SimpleFilter, options);
+
+        let select = $$('select', {className: 'select'});
+        let option = $$('option', {className: 'option'});
+        option.el.innerText = l(this.options.title);
+        select.el.appendChild(option.el) ;
+
+        let overselect = $$('div', {className: 'overselect'});
+        const multiselect = $$('div', { className: 'multiselect'});
+        multiselect.el.appendChild(select.el);
+        multiselect.el.appendChild(overselect.el);
+        this.element.appendChild(multiselect.el);
+
+        this.checkboxContainer = $$('div', { className: 'checkboxes'});
+        this.checkboxes = _.map(this.options.values, (value)=> this.createCheckboxes(value));
+        _.each(this.checkboxes, (result) => {
+            this.checkboxContainer.el.appendChild(result.checkbox.getElement())
+        });
+        this.element.appendChild(this.checkboxContainer.el);
+
+        $$(document.body).on('click', (e: Event)=> {
+            let target = <Element>e.target;
+
+            if( overselect.el == e.target){
+                this.onClick()
+            } else {
+                while (target != this.element) {
+                    if(target != this.root && target.parentElement != null) {
+                        target = target.parentElement;
+                    }
+                    if ((target == this.root || target.parentElement == null) && this.expanded) {
+                        this.onClick();
+                        break;
+                    } else if ((target == this.root || target.parentElement == null) && !this.expanded){
+                        break;
+                    }
+                }
+            }
+        });
+        this.bind.onRootElement(QueryEvents.buildingQuery, (args: IBuildingQueryEventArgs) => this.handleBuildingQuery(args));
+        this.bind.onRootElement(QueryEvents.querySuccess, (args: IQuerySuccessEventArgs) => this.groupBy(args));
+        this.bind.onRootElement(QueryEvents.doneBuildingQuery, (args: IDoneBuildingQueryEventArgs) => this.handleDoneBuildingQuery(args));
+
+    }
+
+    private handleBuildingQuery(args: IBuildingQueryEventArgs) {
+        Assert.exists(args);
+        Assert.exists(args.queryBuilder);
+        let selectedValues = this.getSelectedValues();
+        if(selectedValues.length > 0) {
+            args.queryBuilder.advancedExpression.addFieldExpression(this.options.field.toLocaleString(), '==', selectedValues);
+        }
+    }
+
+    public onClick () {
```

Comment:
the name is confusing.

Call this `toggle`.

Also, we need a `open` and a `close` method